### ISSUE TITLE
Issue with property order

### ISSUE THREAD

**Issue by anshubhatnagar** [agileobjects/AgileMapper#22]
Consider following classes:

```
        public class Country
        {
            public Currency Currency { get; set; }
            public int CurrencyId { get; set; }
        }

        public class Currency
        {
            public int Id { get; set; }
        }
```
Now if we try to do the following:

```
Country country = new Country
{
    CurrencyId = 1
};

Country newCountry = mapper.Map(country).ToANew<Country>();
```

We get ```newCountry.CurrencyId``` as ```0``` instead of ```1```.

But it works fine if we change the order like so;

```
        public class Country
        {
            public int CurrencyId { get; set; }
            public Currency Currency { get; set; }
        }
```

Now we get the correct value as ```1```.

**Comment by SteveWilkes** [comment]
This is fixed in the latest code - release to follow. Thanks for the feedback! :)

**Comment by SteveWilkes** [comment]
This is fixed by [release v0.12](https://www.nuget.org/packages/AgileObjects.AgileMapper/0.12.0). Thanks again for the feedback!
